### PR TITLE
Added deployment unit and integration testing

### PR DESCRIPTION
**Pull request by hpractv**
# Summary of PR changes

## PR Requirements
- [ ] For major, minor, or breaking changes, at least one of the commit messages contains the appropriate `+semver:` keywords.
  - See the *Incrementing the Version* section of the repository's README.md for more details.
- [ ] The action code does not contain sensitive information.

*NOTE: If the repo's workflow could not automatically update the `README.md`, it should be updated manually with the next version.  For javascript actions, if the repo's workflow could not automatically recompile the action it should also be updated manually as part of the PR.*

### CODE REVIEW

File: src/deployments.js [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+  FAILURE: 'failure',
+  INACTIVE: 'inactive',
+  IN_PROGRESS: 'in_progress',
+  QUEUED: 'queued',
+  PENDING: 'pending'
+};
+
+const createOctokitClient = token => new Octokit({ auth: token });
+const createOctokitGraphQLClient = token =>
+  graphql.defaults({ headers: { authorization: `token ${token}` } });
```
Comment (danielle-casella-adams) [inline src/deployments.js:16]: Is there an advantage to defining a function that creates an octokit client as opposed to creating a single client that is passed to the functions that need it?  It seems like having a single client that all the different functions use might be better than each function creating its own version of the client.

File: src/library.test.js [hunk 2]
```diff
@@ -32,13 +31,16 @@ describe('deployment status', () => {
...
-      const testConext = setup();
-      expect(testConext.deployment_status).toBe(status);
-    } catch (error) {
-      expect(error).toBeUndefined();
+  test.each(Object.values(ALLOWED_STATUSES))(
+    `valid deployment status of %s does not throw exception`,
+    status => {
+      process.env[inputKey('deployment-status')] = status;
+      try {
+        const testConext = setup();
```
Comment (danielle-casella-adams) [inline src/library.test.js]: This is a nitpick but it looks like there is a typo here. `testConext` => `testContext`.

Comment (hpractv) [reply src/library.test.js]: 👍🏽

File: src/deployments.js (hunk 1 above)
Comment (hpractv) [reply src/deployments.js:16]: This function is getting used by the deployment testing file, too.  I wanted to make sure the setting params for these clients were the same in both the runtime code as well as the jest test.

**Review by danielle-casella-adams — DISMISSED** [review]

**Review by danielle-casella-adams — APPROVED** [review]

**Pull request merged**